### PR TITLE
versioning support

### PR DESCRIPTION
**Pull request by dtrifiro**
Current usage looks something like this:

```python

def test_on_versioned_s3(tmp_s3_path, versioning):
    assert tmp_s3_path.fs.version_aware
    ...

def test_versioning_generic(tmp_upath_factory):
    tmp_versioned_path = tmp_upath_factory.mktemp("s3", version_aware=True)
    assert tmp_versioned_path.fs.version_aware
``` 

Only `gcs` and `s3` are currently supported.

### CODE REVIEW

**Comment by efiop** [comment]
@dtrifiro Sorry, I accidentally merged `main` into your branch 🙁 Could you rebase, please?

File: src/pytest_servers/fixtures.py [hunk 1]
```diff
@@ -18,9 +18,19 @@
...
+@pytest.fixture
+def versioning():
+    """Enables versioning for supported remotes."""
+    yield
+
+
 @pytest.fixture
 def aws_region_name():
-    return "eu-west-1"
+    return "us-east-1"
```
Comment (efiop) [inline src/pytest_servers/fixtures.py:33]: Is this related to versioning somehow?

**Review by efiop — APPROVED** [review]

File: src/pytest_servers/fixtures.py (hunk 1 above)
Comment (dtrifiro) [reply src/pytest_servers/fixtures.py]: Removed with the latest push

**Comment by rlamy** [comment]
Nice! But how could I write a test that works with both versioned and unversioned buckets?

**Comment by dtrifiro** [comment]
@rlamy 

Something like this will work:

```python
@pytest.mark.parametrize("version_aware", [False, True])
def test_stuff(tmp_upath_factory, version_aware):
    tmp_path = tmp_upath_factory.mktemp("s3", version_aware=version_aware)
    assert tmp_path.fs.version_aware
    ...
```

If that's too clunky, we could modify the versioning fixture so that it works like this:

```python
@pytest.mark.parametrize("versioning", [False, True], indirect=True)
def test_stuff(s3_tmp_path, versioning):
    assert tmp_s3_path.fs.version_aware is versioning
    ...
```

Currently it is enough to use the `versioning` fixture to enable versioning support for the `s3_tmp_path` and `gcs_tmp_path` fixtures.

**Comment by skshetry** [comment]
I don't have strong opinion to this PR, but I think there are two parts to versioning: one on the server side, and other on client-side.

The server already supports this right? And the client side is as simple as following on dql side:
```python
def s3_path():
    return tmp_upath_factory.s3(..., version_aware=True)
```

And this is already possible, right?

**Comment by rlamy** [comment]
@dtrifiro Actually, I'm more interested in making a fixture that handles bucket creation, but I guess something like this works:
```python
@pytest.fixture(scope="session", params=[False, True])
def version_aware(request):
    return request.param

@pytest.fixture
def cloud(tmp_upath_factory, version_aware):
    tmp_path = tmp_upath_factory.mktemp("s3", version_aware=version_aware)
    return do_something(tmp_path)
```
And the parametrisation can be overridden with:
```python
@pytest.mark.parametrize("version_aware", [True], indirect=True)
def test_versioned_only(cloud):
    assert cloud.is_versioned
```
So that works.
